app/webpacker/controllers/bulk_form_controller.js

```diff
@@ -73,7 +74,7 @@ export default class BulkFormController extends Controller {
 
     // Prevent accidental data loss
     if (formChanged) {
-      window.addEventListener("beforeunload", this.preventLeavingBulkForm);
+      window.addEventListener("beforeunload", this.preventLeavingBulkForm); // TOFIX: what if it has laredy been added? we can optimise above to avoid this
```

Comment:
Could this lead to multiple warnings?

I would register a callback only once and then the callback needs to check if the form has changed or not. But the current solution probably works for now.

Reply:
You would think so, but it doesn't. Probably because the browser is smart enough to only ask once. 

That's a really good suggestion though 👍